examples/common/chip-app-server/Server.cpp

```diff
@@ -47,13 +53,79 @@ using namespace ::chip::DeviceLayer;
 DemoSessionManager sessions;
 
 namespace chip {
+
 SecureSessionMgrBase & SessionManager()
 {
     return sessions;
 }
+
 } // namespace chip
 
 namespace {
+
+void HandleBLEConnectionClosed(chip::Ble::BLEEndPoint * endPoint, BLE_ERROR err)
+{
+    if (err != BLE_NO_ERROR)
+    {
+        ChipLogError(AppServer, "BLE Error: %d", err);
+    }
+    ChipLogProgress(AppServer, "BLE Connection closed");
+}
+
+void HandleBLEMessageReceived(chip::Ble::BLEEndPoint * endPoint, chip::System::PacketBuffer * buffer)
+{
+#if CHIP_ENABLE_OPENTHREAD
+    uint16_t bufferLen = buffer->DataLength();
+    uint8_t * data     = buffer->Start();
+    chip::DeviceLayer::Internal::DeviceNetworkInfo networkInfo;
+    ChipLogProgress(AppServer, "Receive BLE message size=%u", bufferLen);
+
+    memcpy(networkInfo.ThreadNetworkName, data, sizeof(networkInfo.ThreadNetworkName));
```

Comment:
This is moving existing code, but shouldn't this have error checking to make sure we have enough data in the buffer for all of this stuff we're doing and that we don't read off the end of the buffer?  Please make sure there's a TODO comment here and a followup issue filed.

Reply:
Buffer length check added.